### PR TITLE
[improve] Add num not adhering placement ledgers replicated metric for ReplicationWorker

### PR DESCRIPTION
**Pull request by wenbingshen**
### Motivation

We have `NUM_FULL_OR_PARTIAL_LEDGERS_REPLICATED` in `ReplicationWorker`, which includes both `DATA_LOSS` and `DATA_NOT_ADHERING_PLACEMENT` data repair types. It is meaningful to add `NUM_NOT_ADHERING_PLACEMENT_LEDGERS_REPLICATED` metric for `DATA_NOT_ADHERING_PLACEMENT` type separately.

### CODE REVIEW

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/TestReplicationWorker.java [hunk 1]
```diff
@@ -1262,6 +1268,9 @@ protected EnsemblePlacementPolicy initializeEnsemblePlacementPolicy(ClientConfig
             assertNull(stat1);
         });
 
+        assertTrue("NUM_NOT_ADHERING_PLACEMENT_LEDGERS_REPLICATED",
```
Comment (zymap) [inline bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/TestReplicationWorker.java:1271]: Please add a new test to make sure the number is increased as expected.

**Review by zymap — CHANGES_REQUESTED** [review]
Overall look good to me. Just need a test.

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/TestReplicationWorker.java (hunk 1 above)
Comment (wenbingshen) [reply bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/TestReplicationWorker.java:1271]: Good suggestion. I found a bug about numLedgersReplicated update. I have submit another PR #3654 to fix it. PTAL. I will update this PR when that PR is merged.

**Comment by hangc0276** [comment]
@horizonzy  Please help take a look at this PR, thanks.

**Comment by horizonzy** [comment]
> @horizonzy Please help take a look at this PR, thanks.

Fine

File: bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/TestReplicationWorker.java (hunk 1 above)
Comment (hangc0276) [reply bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/TestReplicationWorker.java:1271]: +1

**Review by hangc0276 — COMMENTED** [review]
Overall LTGM, thanks for your contribution.

**Review by horizonzy — APPROVED** [review]
LGTM.

**Comment by wenbingshen** [comment]
rerun failure checks

**Comment by wenbingshen** [comment]
rerun failure checks

**Comment by wenbingshen** [comment]
Hi, @zymap @hangc0276 I have updated this PR. Could you help take a look? Thanks.

**Review by zymap — APPROVED** [review]

**Review by hangc0276 — APPROVED** [review]

**Pull request merged**